src/composables/usePSModuleManagement.ts

```diff
@@ -0,0 +1,50 @@
+export enum Action {
+  Enable = 'enable',
+  Install = 'install',
+  Update = 'update',
+}
+
+export enum Module {
+  PsEventbus = 'ps_eventbus',
+  PsAccounts = 'ps_accounts',
+}
+
+export default function usePSModuleManagement(isPs17: boolean) {
+  async function manageModulePS17(module: Module, action: Action, link: string) {
+    try {
+      const response = await fetch(link, {method: 'POST'});
+      const data = await response.json();
+
+      if (!response.ok) {
+        throw new Error(`An error has occured: ${response.status}`);
+      }
+
+      if (data[module].status === false) {
+        throw new Error(`Cannot ${action} ${module} module.`);
+      }
```

Comment:
If we check this condition only when isPs17 is true, we can refactor these two functions into one ?

Reply:
So we will end up with a function like this ?
```ts 
async function manageModule(module: Module, action: Action, link: string) {
    try {
      const response = await fetch(link, {method: 'POST'});
      if(isPs17) {
        const data = await response.json();

        if (!response.ok) {
          throw new Error(`An error has occured: ${response.status}`);
        }

        if (data[module].status === false) {
          throw new Error(`Cannot ${action} ${module} module.`);
        }

        return data;
      }
    } finally {
      //
    }
  }
```